File: praca-inzynierska-backend/application/src/main/kotlin/jwer/pracainzynierskabackend/model/dto/StatisticsDto.kt
package jwer.pracainzynierskabackend.model.dto

import jwer.pracainzynierskabackend.model.entity.Shift

data class StatisticsDto(
        val monthNumber: Int?,
        var shifts: Int = 0,
        var hours: Double = 0.0
) {

    operator fun plusAssign(shift: Shift) {
        shifts++
        hours += shift.period.hours()
    }

    operator fun plusAssign(other: StatisticsDto) {
        shifts += other.shifts
        hours += other.hours
    }

}

File: praca-inzynierska-frontend/src/app/app/model/dto/StatisticsDto.ts
export class StatisticsDto {

    constructor(
        public monthNumber: number,
        public shifts: number,
        public hours: number
    ) {}

    static copyOf(other: StatisticsDto): StatisticsDto {
        return new StatisticsDto(other.monthNumber, other.shifts, other.hours)
    }

}